libraries/chain/hotstuff/finalizer.cpp

```diff
@@ -33,23 +33,25 @@ finalizer::vote_decision finalizer::decide_vote(const block_state_ptr& proposal,
    bool safety_check   = false;
    bool liveness_check = false;
 
-   auto p_branch = fork_db.fetch_branch(proposal->id());
-
    bool monotony_check = !fsi.last_vote || proposal->timestamp() > fsi.last_vote.timestamp;
    // !fsi.last_vote means we have never voted on a proposal, so the protocol feature just activated and we can proceed
 
-   if (!fsi.lock.empty()) {
-      // Safety check : check if this proposal extends the proposal we're locked on
-      // --------------------------------------------------------------------------
-      if (extends(fork_db, proposal, fsi.lock.id))
-         safety_check = true;
+   if (!monotony_check) {
+      dlog("monotony check failed for proposal ${p}, cannot vote", ("p", proposal->id()));
+      return vote_decision::no_vote;
+   }
 
+   if (!fsi.lock.empty()) {
       // Liveness check : check if the height of this proposal's justification is higher
```

Comment:
We don't use height any more. Maybe change to block number?

Reply:
While we standardized on `block_num`, I think using height in this sentence makes it clearer. What do you think @heifner ?

Reply:
Personally, I think height in comments if fine.